dali/operators/image/paste/multipaste.h

```diff
@@ -0,0 +1,200 @@
+
+#ifndef DALI_OPERATORS_IMAGE_PASTE_MULTIPASTE_H_
+#define DALI_OPERATORS_IMAGE_PASTE_MULTIPASTE_H_
+
+#include <limits>
+#include <memory>
+#include <string>
+#include <vector>
+#include "dali/core/static_switch.h"
+#include "dali/kernels/kernel_manager.h"
+#include "dali/pipeline/data/views.h"
+#include "dali/pipeline/operator/common.h"
+#include "dali/pipeline/operator/operator.h"
+#include "dali/pipeline/operator/arg_helper.h"
+#include "dali/core/format.h"
+#include "dali/util/crop_window.h"
+#include "dali/pipeline/data/types.h"
+
+
+namespace dali {
+
+template <typename Backend>
+class MultiPasteOp : public Operator<Backend> {
+ public:
+  ~MultiPasteOp() override = default;
+
+  DISABLE_COPY_MOVE_ASSIGN(MultiPasteOp);
+
+ protected:
+  using Coords = TensorView<typename compute_to_storage<Backend>::type, const int, 1>;
+
+  explicit MultiPasteOp(const OpSpec &spec)
+      : Operator<Backend>(spec)
+      , output_type_arg_(spec.GetArgument<DALIDataType>("dtype"))
+      , output_type_(DALI_NO_TYPE)
+      , input_type_(DALI_NO_TYPE)
+      , output_size_("output_size", spec)
+      , in_idx_("in_ids", spec)
+      , in_anchors_("in_anchors", spec)
+      , in_shapes_("shapes", spec)
+      , out_anchors_("out_anchors", spec) {
+    zero_anchors_ = Coords(zeros_, {2});
+
+    if (std::is_same<Backend, GPUBackend>::value) {
+      kernel_manager_.Resize(1, 1);
+    } else {
+      kernel_manager_.Resize(num_threads_, max_batch_size_);
+    }
+  }
+
+  bool CanInferOutputs() const override {
+    return true;
+  }
+
+  template<int ndim = 2>
+  bool Intersects(const Coords& anchors1, const Coords& shapes1,
+                  const Coords& anchors2, const Coords& shapes2) const {
+    for (int i = 0; i < ndim; i++) {
+      if (anchors1.data[i] + shapes1.data[i] <= anchors2.data[i]
+          || anchors2.data[i] + shapes2.data[i] <= anchors1.data[i]) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  void AcquireArguments(const OpSpec &spec, const workspace_t<Backend> &ws) {
+    const auto &images = ws.template InputRef<Backend>(0);
+
+    auto curr_batch_size = ws.GetRequestedBatchSize(0);
+    output_size_.Acquire(spec, ws, curr_batch_size, true);
+    in_idx_.Acquire(spec, ws, curr_batch_size, false);
+    if (out_anchors_.IsDefined()) {
+      out_anchors_.Acquire(spec, ws, curr_batch_size, false);
+    }
+    if (in_anchors_.IsDefined()) {
+      in_anchors_.Acquire(spec, ws, curr_batch_size, false);
+    }
+    if (in_shapes_.IsDefined()) {
+      in_shapes_.Acquire(spec, ws, curr_batch_size, false);
+    }
+    input_type_ = ws.template InputRef<Backend>(0).type().id();
+    output_type_ =
+        output_type_arg_ != DALI_NO_TYPE
+        ? output_type_arg_
+        : input_type_;
+
+    for (int i = 0; i < curr_batch_size; i++) {
+      raw_input_size_mem_.push_back(static_cast<int>(images.shape()[i].data()[0]));
+      raw_input_size_mem_.push_back(static_cast<int>(images.shape()[i].data()[1]));
+    }
```

Comment:
Multiple bugs here:
* raw_input_size_mem_ is not cleared and stale data will remain at the beginning, while new data will be added to the end indefinitely
* at some point, input shapes may have a different batch size than output - use num_samples from the relevant shape whenever it's available
```suggestion
    raw_input_size_mem_.clear();
    auto &input_shape = images.shape();
    raw_input_size_mem_.reserve(spatial_ndim * input_shape.num_samples());
    for (int i = 0; i < curr_batch_size; i++) {
      for (int extent : images.shape().tensor_shape_span(i))
        raw_input_size_mem_.push_back(extent);
    }
```

Reply:
For some reason, when using this syntax, in the first loop (`i = 0`) the values of `extent` were random for width and heigth, as if the first call of `tensor_shape_span` returned garbage. Values in next iterations (i > 0) and the number of channels (which is after should not be stored in `raw_input_size_mem_`) the values were correct. 

Used following syntax in the loop and it seems is working correctly:
```
for (int i = 0; i < curr_batch_size; i++) {
      auto shape = input_shape.tensor_shape_span(i);
      for (int j = 0; j < spatial_ndim; j++) {
        auto extent = static_cast<int>(shape[j]);
        raw_input_size_mem_.push_back(extent);
      }
    }
```